samples/grids/grid/infinite-scroll/src/index.tsx

```diff
@@ -1,89 +1,114 @@
-import React, { useEffect, useRef, useState } from 'react';
+import React, { useEffect, useRef } from 'react';
 import ReactDOM from 'react-dom/client';
 import './index.css';
 
-import { IgrForOfStateEventArgs, IgrGridModule } from 'igniteui-react-grids';
+import { IgrForOfState } from 'igniteui-react-grids';
 import { IgrGrid, IgrColumn } from 'igniteui-react-grids';
 import { loadDataForPage, getCachedData } from './NwindService';
 
 import 'igniteui-react-grids/grids/combined';
 import 'igniteui-react-grids/grids/themes/light/bootstrap.css';
 
-const mods: any[] = [
-    IgrGridModule
-];
-mods.forEach((m) => m.register());
 
 export default function App() {
     let pageSize = 10;
     let page = 1;
     let totalPageCount = 0;
     let totalItems = 0;
     const gridRef = useRef<IgrGrid>(null);
-
+  
     useEffect(() => {
+      gridRef.current.isLoading = true;
+      const dataViewSize = 9.6;
+      pageSize = Math.floor(dataViewSize * 1.5);
+  
+      loadDataForPage(page, pageSize, (request) => {
+        if (request.data) {
+          gridRef.current.data = getCachedData({ startIndex: 0, chunkSize: 10 });
+          gridRef.current.totalItemCount = page * pageSize;
+          totalItems = request.data["@odata.count"];
+          totalPageCount = Math.ceil(totalItems / pageSize);
+          gridRef.current.isLoading = false;
+        }
+      });
+    }, []);
+  
+    function handlePreLoad(e: IgrForOfStateEventArgs) {
```

Comment:
and `IgrForOfStateEventArgs` isn't imported, possibly typo on the unused IgrForOfState import? If so, you can change the import name